### ISSUE TITLE
Trakt integration and PIN-code authentication not working anymore

### ISSUE THREAD

**Issue by CptChaos** [SickRage/SickRage#1363]
I got warnings in Sick Rage about Trakt. I tested Trakt and it failed.
As a test, I wanted to reconnect Sick Rage with Trakt, so I revoked Sick Rage's access to my Trakt account. I wanted to reconnect, but I can't reconnect anymore, as per an API update on their side, the PIN-code authentication method is now deprecated. The page https://trakt.tv/pin for example isn't available anymore. Please update Sick Rage to use latest authentication methods. :)

See also their API docs:
http://docs.trakt.apiary.io/#reference/authentication-pin

**Comment by SickRage2** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [ ] Include basic information: Branch/Commit, OS, What you did, What happened, What you expected
- [ ] Enable debug logging (be sure to disable after the bug is fixed)
- [ ] Post debug logs, either inline (for smaller logs) or using gist

Please make sure you also read [how to create an issue](https://github.com/SickRage/sickrage-issues/blob/master/README.md) and followed all of the steps. 

The title should describe your issue. Having "SR not working" or "I get this bug" for 100 issues, isn't really helpful. We will close issues if there isn't enough information.

Sometimes the devs may seem like grunts and respond with short answers. This isn't (always) because the dev hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let us know, and this bot may get updated to automatically answer you.

Thanks!

**Comment by CptChaos** [comment]
Not that it's mandatory for this issue;
I'm using the master branch, on a Ubuntu 15.10 system.

Warnings I got are:

> 
> 2016-04-07 00:05:11 Thread-14 :: Unauthorized. Please check your Trakt settings
> 2016-04-06 23:34:26 TRAKTCHECKER :: Unauthorized. Please check your Trakt settings
> 2016-04-06 12:30:47 Thread-13 :: Unauthorized. Please check your Trakt settings
> 2016-04-06 12:30:16 SEARCHQUEUE-BACKLOG-272644 :: Unauthorized. Please check your Trakt settings
> 2016-04-06 12:28:59 SEARCHQUEUE-BACKLOG-301824 :: Unauthorized. Please check your Trakt settings
> 2016-04-06 10:51:18 SEARCHQUEUE-DAILY-SEARCH :: Unauthorized. Please check your Trakt settings

**Comment by binarylimited** [comment]
I noticed that too.
From what I saw, everytime I did an update, it would lose the login details and I would have to input the sick rage login details again.

**Comment by duramato** [comment]
Working fine here, for the Get Trakt PIN to appear you need to put some dummy letter on the text box first and the button should appear you open the page, get the PIN and click autorize.

**Comment by CptChaos** [comment]
@duramato Don't know what you mean. Also, PIN-code authentication is deprecated already in the Trakt v2 API, so expect it to fully disappear any time. Would be nice if Sick Rage would be ahead of that.

**Comment by CptChaos** [comment]
As posted 5 days ago, I managed to reauthenticate Trakt with my Sick Rage install, yet I still get these errors:

2016-04-14 19:30:54 TRAKTCHECKER :: Unauthorized. Please check your Trakt settings
2016-04-13 19:06:58 Thread-13 :: Unauthorized. Please check your Trakt settings
2016-04-13 19:06:04 SEARCHQUEUE-DAILY-SEARCH :: Unauthorized. Please check your Trakt settings

Seems like I need to reauthorize again, due to PIN authentication.

**Comment by neoatomic** [comment]
As it seems to work, i'm going to close this for now.
Its probbebly something we need to look at if we switch/implement the Trakt v2 API.

**Comment by CptChaos** [comment]
I don't agree upon closing this issue, as the authentication API being used is deprecated on Trakt API.
Also the way to authenticate Sick Rage on Trakt isn't done that easily anymore. Would be nice if Sick Rage switched to the proper un-deprecated method(s) for authentication with Trakt API v2.

**Comment by neoatomic** [comment]
Its on the To-Do list. But rewriting the Trakt client isn't something that can be done on a short notice. 
However we are open for a Pull Request if you are able to help out.

**Comment by AbhieSpeaks** [comment]
Any updates on this issue. Been a year since this was requested.

**Comment by CptChaos** [comment]
I would like to bring this issue under the attention again. By the end of May, Trakt.tv will be stopping with authorization code, which SR currently uses. If I noticed the post earlier, I would've brought this up much sooner.

**Comment by msyquest** [comment]
Do we have any update on this issue?

**Comment by billbo99** [comment]
thanks for re-opening the ticket,  I can am also having the same issue.   For now I have turned off the service within SickRage while I watch this ticket for an update

2017-07-15 17:57:06 WARNING TRAKTCHECKER :: Unauthorized. Please check your Trakt settings
2017-07-15 17:57:06 WARNING TRAKTCHECKER :: You must get a Trakt TOKEN. Check your Trakt settings

**Comment by sharkykh** [comment]
I spent 2 hours yesterday going at this (updating the code to use OAuth2) but it would take me some more time to finish this.

Having said that, I don't understand the issue.

* SickRage **is using API v2**. Granted, it uses PIN auth which has been deprecated for some time now, but **it still works.**
* @CptChaos The code **does not** use the "old token authentication" system (username/password login). Also, read the comments in the Trakt API community - PIN auth is still supported with no plan of removing support for it just yet, because it is OAuth.
* I've been using Trakt in SickRage for couple of months now and haven't had any issue.
  Also, I just authed Trakt from scratch (removed tokens and re-authed) yesterday, and it worked.

So, what is the issue exactly?

**Comment by msyquest** [comment]
Hello Sharkykh,
Thank you for your time :)

In SickRage, when I hit "Test Trakt" it confirm me that "Test notice sent successfully to Trakt" but the log eventually give me some of these :
"
2017-07-16 18:09:58 Thread-15 :: You must get a Trakt TOKEN. Check your Trakt settings
2017-07-16 18:09:58 Thread-15 :: Unauthorized. Please check your Trakt settings
2017-07-16 17:41:54 TRAKTCHECKER :: You must get a Trakt TOKEN. Check your Trakt settings
2017-07-16 17:41:54 TRAKTCHECKER :: Unauthorized. Please check your Trakt settings
2017-07-16 03:14:07 POSTPROCESSOR-AUTO :: You must get a Trakt TOKEN. Check your Trakt settings
2017-07-16 03:14:07 POSTPROCESSOR-AUTO :: Unauthorized. Please check your Trakt settings
2017-07-16 03:09:38 SEARCHQUEUE-DAILY-SEARCH :: You must get a Trakt TOKEN. Check your Trakt settings
2017-07-16 03:09:38 SEARCHQUEUE-DAILY-SEARCH :: Unauthorized. Please check your Trakt settings
"

Te main issue I can see is the "Sync libraries" and "Sync watchlist" feature that doesn't sync anymore.

SickRage Info:
Branch: master
Commit: 96b53dee713dc35ebf295d0f37a67b2a36fb4ea3
Version: v2017.06.05-1
Database Version: 44.0
  Python Version: 2.7.9 (default, Jun 29 2016, 13:08:31) [GCC 4.9.2]
  SSL Version: OpenSSL 1.0.1t 3 May 2016
  OS: Linux-3.16.0-4-amd64-x86_64-with-debian-8.7
  Locale: en_GB.UTF-8

Do you need more information?

**Comment by msyquest** [comment]
As for authed Trakt from scratch (removed tokens and re-authed), I'm not so sure on how to do it...

**Comment by sharkykh** [comment]
> As for authed Trakt from scratch (removed tokens and re-authed), I'm not so sure on how to do it...

It the same as clicking "Get Token" and authenticating with a new code.
If you can't see "Get Token" just type anything in the "Trakt PIN" text box, then delete it. It should show up.

**Comment by msyquest** [comment]
Hello,

> It the same as clicking "Get Token" and authenticating with a new code.
> If you can't see "Get Token" just type anything in the "Trakt PIN" text box, then delete it. It should show 
> up.

Good news, I did that and the sync started to work again.

Thank you very much!

**Comment by CptChaos** [comment]
@sharkykh I've never said SR used old token system with username/password, all I said was SR uses deprecated API's for authentication with Trakt. When I opened this issue over a year ago, the API being used (by SR) for authentication was deprecated.

A month ago, it's dev made this post (https://plus.google.com/109991097847241028373/posts/SX8XS8V2ocz), saying PIN is also deprecated, but would not get removed soon as it's OAuth based. It nice and all that it's still works, but it's deprecated non the less; eventually it will get removed.

I don't like the idea a tool uses deprecated (auth) API's for security reasons. If I'd knew programming in Python, I would gladly help and make a PR request, but I don't, so I made an issue instead.

**Comment by sharkykh** [comment]
@CptChaos Ok, so I must have misunderstood you.

But I still don't understand where these errors come from.
> 2016-04-18 22:50:22 TRAKTCHECKER :: You must get a Trakt TOKEN. Check your Trakt settings
> 2016-04-18 22:50:22 TRAKTCHECKER :: Unauthorized. Please check your Trakt settings

I only encounter them when my Trakt access token isn't valid.
Perhaps it's not refreshing it correctly...
Does it take some time for these errors to show up again after you re-authenticate Trakt in SickRage?

**Comment by CptChaos** [comment]
That's the error SR returned. :)

**Comment by AzureCerulean** [comment]
new URL https://trakt.tv/activate
